src/ocean/core/SmartUnion.d

```diff
@@ -306,11 +307,11 @@ version ( UnitTest )
 
 *******************************************************************************/
 
-private const handleInvalidCases = "case none: assert(false);" ~
+private const handleInvalidCases = "case none: verify(false); break;" ~
     // The D2 build makes sure all enum cases are covered so in fact we don't
     // need a default. However, D1 emits a compiler warning if a switch
     // has no default so we add a default here just to avoid the warning.
-    "version (D_Version2){} else {default: assert(false);}";
+    "version (D_Version2){} else {default: verify(false); break; }";
```

Comment:
Well I'd be inclined to keep the `assert` here, as this case is *truly* unreachable. And it's never compiled in D2 anyway.

Reply:
I did think about it and this case makes me sad. While it's true it's truly unreachable, it might not be for the D1-only compiled application. In which case, it doesn't matter if it's `AssertException` or `SanityException`, but I've changed it only for the consistency (as it's not compiled in D2 anyway). I don't know, let's see what other think, I don't feel right about this change, I'll tell you that.

Reply:
> While it's true it's truly unreachable, it might not be for the D1-only compiled application.

This is a contradiction, and I think it is truly unreachable, the D2 build makes sure of that. Or when do you think it could be reachable in a D1 build?
Here "truly unreachable" does not mean there is a formal proof, instead it means there is a guarantee -- the D2 build -- that each enum member has a `case`. Even D2 `final switch` doesn't prevent `u.active` from returning a value outside the range of the enum members.

Reply:
Yes, I was thinking about cases when there's no D2 build for the project. But then, they shouldn't use `handleInvalidCases` anyway. I'll just revert this change :+1: